src/test/ui/feature-gates/issue-43106-gating-of-builtin-attrs.rs

```diff
@@ -298,35 +298,32 @@ mod path {
     mod inner { #![path="3800"] }
 
     #[path = "3800"] fn f() { }
-    //~^ WARN unused attribute
+    //~^ WARN `#[path]` only has an effect
 
     #[path = "3800"]  struct S;
-    //~^ WARN unused attribute
+    //~^ WARN `#[path]` only has an effect
 
     #[path = "3800"] type T = S;
-    //~^ WARN unused attribute
+    //~^ WARN `#[path]` only has an effect
 
     #[path = "3800"] impl S { }
-    //~^ WARN unused attribute
+    //~^ WARN `#[path]` only has an effect
 }
 
+// Don't warn on `automatically_derived` - a custom derive
+// could reasonally annotate anything that it emits with
+// this attribute
```

Comment:
This attribute is pretty under-specified - it seems to suppress a handful of lints, which aren't documented anywhere. I think the best approach would be to warn when it's used on a non-impl (since impls are the only places where we currently check for it).